### PR TITLE
misc/byteorder.h: Add helpers to get & put 24-bit unaligned integers

### PR DESCRIPTION
**Pull request by cvinayak**
Added helpers to read and write buffers containing 24-bit
integers with a specific endianness.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

### CODE REVIEW

**Review by carlescufi — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Review by wopu-ot — APPROVED** [review]
LGTM

I wondered a bit about the omission of sys_cpu_to_be24/sys_cpu_to_le24/... but given the lack of support for 24 bit types in C, I believe the PR does what it should do.

File: include/sys/byteorder.h [hunk 1]
```diff
@@ -139,6 +139,21 @@ static inline void sys_put_be16(u16_t val, u8_t dst[2])
 	dst[1] = val;
 }
 
+/**
+ *  @brief Put a 24-bit integer as big-endian to arbitrary location.
```
Comment (MaureenHelm) [inline include/sys/byteorder.h]: Perhaps mention that the 24-bit integer is right justified?

**Review by nashif — CHANGES_REQUESTED** [review]
Need test coverage, see #16942

File: include/sys/byteorder.h (hunk 1 above)
Comment (cvinayak) [reply include/sys/byteorder.h]: I did not quite get by right justification. You mean, the source is right justified in the CPU endian u32_t that gets put into 3-byte destination?

**Comment by cvinayak** [comment]
> Need test coverage, see #16942

Will add

File: include/sys/byteorder.h (hunk 1 above)
Comment (MaureenHelm) [reply include/sys/byteorder.h]: A 24-bit quantity in a 32-bit register could either be right justified (0x00FF_FFFF) or left-justified (0xFFFF_FF00). Left justification is sometimes used for fixed-point fractional numbers

Comment (cvinayak) [reply include/sys/byteorder.h]: Agreed. Will update the comments.

**Comment by cvinayak** [comment]
Closing this as superseded by #20633

**Pull request closed without merging**